.github/workflows/push.yml

```diff
@@ -0,0 +1,97 @@
+name: Cypress Tests
+
+on: [push]
+
+env:
+  ESLINT_NO_DEV_ERRORS: true
+  TSC_COMPILE_ON_ERROR: true
+  IMAGE_INLINE_SIZE_LIMIT: 0
+  INLINE_RUNTIME_CHUNK: false
+  REACT_APP_HIDE_SPLASH: false
+  REACT_APP_UNCHAINED_ETHEREUM_HTTP_URL: https://dev-api.ethereum.shapeshift.com
+  REACT_APP_UNCHAINED_ETHEREUM_WS_URL: wss://dev-api.ethereum.shapeshift.com
+  REACT_APP_UNCHAINED_BITCOIN_HTTP_URL: https://dev-api.bitcoin.shapeshift.com
+  REACT_APP_UNCHAINED_BITCOIN_WS_URL: wss://dev-api.bitcoin.shapeshift.com
+  REACT_APP_PORTIS_DAPP_ID: 8609e6a8-e0dc-45e6-a0ad-edde63a4cdda
+  REACT_APP_ETHEREUM_NODE_URL: https://mainnet.infura.io/v3/d734c7eebcdf400185d7eb67322a7e57
+  REACT_APP_METAMASK_DEEPLINK_URL: https://metamask.app.link/dapp/app.shapeshift.com
+
+jobs:
+  test-chrome:
+    name: Run Cypress
+    runs-on: ubuntu-20.04
+    strategy:
+      # When one test fails, do not cancel the other
+      # containers, because this will kill Cypress processes,
+      # leaving the Dashboard hanging: https://github.com/cypress-io/github-action/issues/48
+      fail-fast: false
+      matrix:
+        # Run 3 copies of the current job in parallel
+        # We can continue adding more as our test suite grows
+        containers: [1, 2, 3]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # Restore the previous yarn modules and Cypress binary archives.
+      # Any updated archives will be saved automatically after the entire
+      # workflow successfully finishes.
+      # We use exact restore key to avoid yarn module snowballing
+      # https://glebbahmutov.com/blog/do-not-let-npm-cache-snowball/
+      - name: Cache central yarn modules
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/yarn
+          key: ${{ runner.os }}-node-${{ github.ref }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-${{ github.ref }}-${{ hashFiles('**/package-lock.json') }}
+
+      # We use the exact restore key to avoid Cypress binary snowballing
+      # https://glebbahmutov.com/blog/do-not-let-cypress-cache-snowball/
+      - name: Cache Cypress binary
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/Cypress
+          key: cypress-${{ runner.os }}-cypress-${{ github.ref }}-${{ hashFiles('**/package.json') }}
+          restore-keys: |
+            cypress-${{ runner.os }}-cypress-${{ github.ref }}-${{ hashFiles('**/package.json') }}
+
+      # Cache local node_modules to pass to testing jobs
+      - name: Cache local node_modules
+        uses: actions/cache@v2
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-node-modules-${{ github.ref }}-${{ hashFiles('**/package-lock.json') }}
```

Comment:
Is `runner.os` interpolation in key necessary across this file since it's going to be the same across runs?

Reply:
Initially it will always be the same, though with the parallelization method used we have the option to test across multiple OSs from this job, e.g. maybe we need a MacOS build for some reason (we can even randomise it, for better test coverage).

This implementation supports such tests architecture in the future.

Reply:
To extend that thought a bit, and add a little more context on its current utility: if we ever changed the OS of the runner, or it was automatically upgraded because we are using "latest" - if this string were not a function of the OS we'd unknowingly end up with an invalid cache, which wouldn't be regenerated, because the key wouldn't change.